File: Java/src/main/java/Solution501_600/Solution520.java
package Solution501_600;

/**
 * @author windows
 */
public class Solution520 {
    public boolean detectCapitalUse(String word) {
        return word.toUpperCase().equals(word)
                || word.toLowerCase().equals(word)
                || (Character.isLetter(word.charAt(0) + 32) && word.substring(1).equals(word.substring(1).toLowerCase()));
    }


    public static void main(String[] args) {
        Solution520 solution520 = new Solution520();
        System.out.println(solution520.detectCapitalUse("Leetcode"));
    }
}

File: C++/src/Solution501_600/Solution520.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution520.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool Solution520::detectCapitalUse(string word) {
    int n = word.length();
    if (word[0] >= 'a'){
        for (int i = 1; i < n; ++i) {
            if (word[i] < 'a'){
                return false;
            }
        }
    } else{
        if (word[1] >= 'a'){
            for (int i = 2; i < n; ++i){
                if (word[i] < 'a'){
                    return false;
                }
            }
        } else{
            for (int i = 2; i < n; ++i){
                if (word[i] >= 'a'){
                    return false;
                }
            }
        }
    }
    return true;
}
